File: php/testcode/ssti_018.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssti_smarty_string_display
function ssti018(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('tpl');
    $smarty = new Smarty();
    $smarty->display('string:' . $input); // vuln-code-snippet vuln-line php_ssti_smarty_string_display
    return BenchmarkResponse::ok('rendered');
}
// vuln-code-snippet end php_ssti_smarty_string_display

File: ruby/testcode/ssti_018.rb
require_relative 'shared'

begin
  require 'haml'
rescue LoadError
end

# vuln-code-snippet start ruby_ssti_haml_render_string
def handler(req)
  output = Haml::Engine.new(req.param('template')).render # vuln-code-snippet vuln-line ruby_ssti_haml_render_string
  BenchmarkResponse.html(output)
end
# vuln-code-snippet end ruby_ssti_haml_render_string
